ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -521,19 +539,126 @@ def _run(self) -> Tuple[List[EventOddsRatio], bool]:
+        params = self._filter.with_data(
+            {
+                "funnel_correlation_person_converted": "true" if success else "false",
+                "funnel_correlation_person_entity": json.dumps({"id": event_definition["event"], "type": "events"}),
+            }
+        ).to_params()
+        return f"/api/person/funnel/correlation/?{urllib.parse.urlencode(params)}"
+
+    def construct_event_with_properties_people_url(self, success: bool, event_definition: EventDefinition) -> str:
+        if self.support_autocapture_elements():
+            # If we have an $autocapture event, we need to special case the
+            # url by converting the `elements` chain into an `Action`
+            event_name, _, _ = event_definition["event"].split("::")
+            elements = event_definition["elements"]
+            first_element = elements[0]
+            elements_as_action = {
+                "tag_name": first_element["tag_name"],
+                "href": first_element["href"],
+                "text": first_element["text"],
+                "selector": build_selector(elements),
+            }
+            params = self._filter.with_data(
                 {
-                    "success_count": odds_ratio["success_count"],
-                    "failure_count": odds_ratio["failure_count"],
-                    "odds_ratio": odds_ratio["odds_ratio"],
-                    "correlation_type": odds_ratio["correlation_type"],
-                    "event": self.serialize_event_with_property(odds_ratio["event"]),
+                    "funnel_correlation_person_converted": "true" if success else "false",
+                    "funnel_correlation_person_entity": json.dumps(
+                        {
+                            "id": event_name,
+                            "type": "events",
+                            "properties": [
+                                {
+                                    "key": property_key,
+                                    "value": [property_value],
+                                    "type": "element",
+                                    "operator": "exact",
+                                }
+                                for property_key, property_value in elements_as_action.items()
+                                if property_value
+                                if not None
+                            ],
+                        }
+                    ),
                 }
-                for odds_ratio in results[0]
-            ],
-            "skewed": results[1],
+            ).to_params()
+            return f"/api/person/funnel/correlation/?{urllib.parse.urlencode(params)}"
+
+        event_name, property_name, property_value = event_definition["event"].split("::")
+        params = self._filter.with_data(
+            {
+                "funnel_correlation_person_converted": "true" if success else "false",
+                "funnel_correlation_person_entity": json.dumps(
```

Comment:
Q: If `with_data` is working now, shouldn't the `json.dumps` bit happen inside `to_params` ?

Reply:
`with_data` is setting the data as expected by from the API, which is json encoded. I think this will break the interface if we do not pass in encoded?

Reply:
Ahhh, right, my bad!

Reply:
Actually: all filters support both: json.dump or a dict object.

The general flow seems to be: (1) json.dumps when passing via URL, and (2) pass in object when using `with_data`.

So, I think not dumping works fine here as well https://github.com/PostHog/posthog/blob/master/posthog/models/filters/mixins/funnel.py#L327

Don't mind either though.